### PR TITLE
TACo Mainnet deployment parameters and intermediate artifacts

### PR DESCRIPTION
**Pull request by cygnusv**
(Depends on #196, wait for it to be merged)

Specially important for reviewers:
* FxPortal Deployment info for Mainnet: FxRoot, FxChild, CheckpointManager (See https://wiki.polygon.technology/docs/pos/design/bridge/l1-l2-communication/state-transfer/#prerequisites)
* DAI address on Polygon: https://polygonscan.com/token/0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063
* Initial fee rate, which is 35 cents per day but expressed in DAI units per second. In Python:
```py
>>> 35*10**16//86400
4050925925925
```
-----

**Update**: We used these parameters to deploy TACo contracts on Ethereum Mainnet and Polygon Mainnet. Intermediary artifacts are included in this PR, as we need additional verification before confirming this is the official deployment.

### CODE REVIEW

File: deployment/constructor_params/mainnet/mainnet-child-params.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "PolygonChild": {},
+  "ProxyAdmin": {},
+  "LynxTACoChildApplication": {
```
Comment (derekpierre) [inline deployment/constructor_params/mainnet/mainnet-child-params.json:4]:
```suggestion
  "TACoChildApplication": {
```

**Review by theref — APPROVED** [review]

**Comment by vzotova** [comment]
Just a reminder, missing 2 parameters: `TACoChildApplication.minimumAuthorization` and `TACoApplication.commitmentDeadline`

File: deployment/constructor_params/mainnet/child.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  PRIVATE_BETA_FEE_RATE: 4050925925925  # $0.35 per day, expressed in DAI units per seconds (in Python: 35*10**16 // 86400)
+  MAX_DKG_SIZE: 32
+  DKG_TIMEOUT_IN_SECONDS: 3600  # One hour in seconds
+  FORTY_THOUSAND_TOKENS_IN_WEI_UNITS: 40000000000000000000000
+
+  # Threshold Network - References:
+  # - https://docs.threshold.network/resources/contract-addresses/mainnet/threshold-dao
+  TREASURY_GUILD_ON_POLYGON: "0xc3Bf49eBA094AF346830dF4dbB42a07dE378EeB6"
+  INTEGRATIONS_GUILD_ON_POLYGON: "0x5bD70E385414C8dCC25305AeB7E542d8FC70e667"
+  THRESHOLD_COUNCIL_ON_POLYGON: "0x9F6e831c8F8939DC0C830C6e492e7cEf4f9C2F5f"
```
Comment (piotr-roslaniec) [inline deployment/constructor_params/mainnet/child.yml:25]: These are described as "Threshold DAO Contracts (Ethereum L1)". Is it a typo? These are supposed to be `_ON_POLYGON`, right?

Comment (cygnusv) [reply deployment/constructor_params/mainnet/child.yml:25]: Yes, I'm afraid there's no actual public documentation about Threshold addresses on Polygon. 
@Luna5-threshold can you help here or bug someone in the DAO about this?

File: deployment/constructor_params/mainnet/root.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  # TACo specific constants:
+  IN_SECONDS_1_HOUR: 3600
+  IN_SECONDS_1_DAY: 86400
+  IN_SECONDS_7_DAYS: 604800
+  IN_SECONDS_60_DAYS: 5184000
+  IN_SECONDS_182_DAYS: 15724800
+  IN_SECONDS_364_DAYS: 31449600
+  FORTY_THOUSAND_TOKENS_IN_WEI_UNITS: 40000000000000000000000
+  TIMESTAMP_FOR_2024_01_01_0000_UTC: 1704067200  # FIXME: Made up value, waiting for https://github.com/nucypher/nucypher/issues/3342
```
Comment (manumonti) [inline deployment/constructor_params/mainnet/root.yml:32]:
```suggestion
  TIMESTAMP_FOR_2024_01_01_0000_UTC: 1703930400
```

Based on https://github.com/nucypher/nucypher/issues/3342#issuecomment-1821205103

December 30th at midnight UTC-11

**Review by manumonti — CHANGES_REQUESTED** [review]

File: deployment/constructor_params/mainnet/root.yml (hunk 3 above)
Comment (arjunhassard) [reply deployment/constructor_params/mainnet/root.yml:32]:
```suggestion
  TIMESTAMP_FOR_2023_12_31_1100_UTC: 1703930400  # FIXME: Made up value, waiting for https://github.com/nucypher/nucypher/issues/3342
```
technically someone in the kiribati islands can game this and commit one hour into january 1st 2024, but honestly good for them

Comment (derekpierre) [reply deployment/constructor_params/mainnet/root.yml:32]: FYI
```
In [4]: m = maya.MayaDT(epoch=1703930400)

In [5]: m.rfc3339()
Out[5]: '2023-12-30T10:00:00.0Z'
```

File: scripts/mainnet/deploy_child.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+
+from ape import project
+
+from deployment.constants import (
+    CONSTRUCTOR_PARAMS_DIR,
+    OZ_DEPENDENCY,
+)
```
Comment (manumonti) [inline scripts/mainnet/deploy_child.py:8]: not a relevant thing, but OZ_DEPENDENCY is not actually used.

```suggestion
from deployment.constants import CONSTRUCTOR_PARAMS_DIR
```

**Review by manumonti — CHANGES_REQUESTED** [review]

File: deployment/constructor_params/mainnet/root.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  FXPORTAL_FXROOT: "0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2"
+  POLYGON_CHILD: "invalid" # Temporary invalid value that will be overwritten by workaround in deployment script
+
+  # TACo specific constants:
+  IN_SECONDS_1_DAY: 86400
+  IN_SECONDS_182_DAYS: 15724800
+  IN_SECONDS_364_DAYS: 31449600
+  IN_SECONDS_AVERAGE_MONTH_DURATION: 2628000  # 365*24*60*60/12
+  FORTY_THOUSAND_TOKENS_IN_WEI_UNITS: 40000000000000000000000
+  TIMESTAMP_FOR_2023_12_31_1100_UTC: 1703930400
```
Comment (manumonti) [inline deployment/constructor_params/mainnet/root.yml:30]: As Derek mentioned, the timestamp `1703930400` corresponds to 2023 12 31 10:00 UTC. So we should change the variable's name
```suggestion
  TIMESTAMP_FOR_2023_12_31_1000_UTC: 1703930400
```

**Review by manumonti — CHANGES_REQUESTED** [review]

File: deployment/constructor_params/mainnet/root.yml (hunk 5 above)
Comment (cygnusv) [reply deployment/constructor_params/mainnet/root.yml:30]: Are we 100% sure that's the timestamp we want?

File: deployment/constructor_params/mainnet/child.yml (hunk 2 above)
Comment (cygnusv) [reply deployment/constructor_params/mainnet/child.yml:25]: https://github.com/keep-network/tbtc-v2/issues/594

I think this is good enough, from tBTCv2 cross-chain deployment, when the Council MS on Polygon was created.

**Review by vzotova — APPROVED** [review]

**Review by theref — APPROVED** [review]

**Review by manumonti — APPROVED** [review]

**Review by piotr-roslaniec — APPROVED** [review]

**Pull request merged**